platform/viewer/src/connectedComponents/ToolbarRow.js

```diff
@@ -150,46 +150,98 @@ class ToolbarRow extends Component {
   }
 }
 
-/**
- * Determine which extension buttons should be showing, if they're
- * active, and what their onClick behavior should be.
- */
-function _getButtonComponents(toolbarButtons, activeButtons) {
-  return toolbarButtons.map((button, index) => {
-    let activeCommand = undefined;
-
-    if (button.buttons && button.buttons.length) {
-      // Iterate over button definitions and update `onClick` behavior
-      const childButtons = button.buttons.map(childButton => {
-        childButton.onClick = _handleToolbarButtonClick.bind(this, childButton);
+function _getCustomButtonComponent(button, activeButtons) {
+  const CustomComponent = button.CustomComponent;
 
-        if (activeButtons.indexOf(childButton.id) > -1) {
-          activeCommand = childButton.id;
-        }
+  function isFunctionalComponent(Component) {
+    return (
+      typeof Component === 'function' &&
+      !(Component.prototype && Component.prototype.isReactComponent)
+    );
+  }
 
-        return childButton;
-      });
+  function isClassComponent(Component) {
+    return !!(
+      typeof Component === 'function' &&
+      Component.prototype &&
+      Component.prototype.isReactComponent
+    );
+  }
```

Comment:
I don't know that we care if it's a functional component or a class component. We only really need it to be defined, and we can assume that if it is, that they've supplied the right type of arg. These could both be swapped with:

`const shouldUseCustomComponent = typeof CustomComponent === 'function';`

Reply:
ok